### PR TITLE
[FLINK-27443] Create standalone mode parameters and decorators for JM and TMs

### PR DESCRIPTION
**Pull request by usamj**
This PR creates decorators and parameters which contain information needed to create JM and TM deployments in standalone mode. These will then be used by the Standalone ClusterDescriptor to actually create the resources needed for a standalone Flink Cluster. 

The `flink-kubernetes-standalone-cluster` is introduced as the parameters and decorators integrate with their similar counterparts in the main Flink package which use a different Fabric8 version that is shaded. 

Note: This is a draft as the FLIP is still under discussion

### CODE REVIEW

**Comment by morhidi** [comment]
@wangyang0918 @gyfora @mbalassi can you please kick the WF process, it's pending approval? Shall we wait for the 1.0.0 release cut before adding this feature?

**Comment by gyfora** [comment]
@usamj i think we need to vote on the flip before we can start with this properly

**Comment by gyfora** [comment]
Could you start the voting thread?

**Comment by usamj** [comment]
Sure, I've created the voting thread

File: flink-kubernetes-standalone-cluster/src/main/java/org/apache/flink/kubernetes/operator/kubeclient/decorators/UserLibMountDecorator.java
Comment (gyfora) [inline flink-kubernetes-standalone-cluster/src/main/java/org/apache/flink/kubernetes/operator/kubeclient/decorators/UserLibMountDecorator.java:39]: Wouldn't this simply come from the user provided image?

**Review by gyfora — COMMENTED** [review]
In general there seem to be some overlap with the decorators that already exist in Flink. It will be good to see this in use :) 

I left some minor comments.

File: flink-kubernetes-standalone-cluster/src/main/java/org/apache/flink/kubernetes/operator/kubeclient/decorators/UserLibMountDecorator.java
Comment (usamj) [reply flink-kubernetes-standalone-cluster/src/main/java/org/apache/flink/kubernetes/operator/kubeclient/decorators/UserLibMountDecorator.java:39]: This can come from user provided image but the images in the flink docker don't have this directory created. The reason it needs to be created is that if it isn't created then configuration defined user classpaths are ignored (https://github.com/apache/flink/blob/b904c948fb17f57b81a8e794cc67718c432bbcaa/flink-clients/src/main/java/org/apache/flink/client/program/DefaultPackagedProgramRetriever.java#L119)

**Review by gyfora — CHANGES_REQUESTED** [review]
Added a minor comment to simplify module naming, and after that I think we can merge this

**Comment by usamj** [comment]
Updated the module names

**Comment by gyfora** [comment]
This can be closed I think, it is included in your other PR

**Pull request closed without merging**
